File: backend/domain/task.go
package domain

type NullableString struct {
	Value  string
	IsNull bool
}

type NullableInteger struct {
	Value  int
	IsNull bool
}

func PtrToNullInt(p *int) NullableInteger {
	if p == nil {
		return NullableInteger{
			IsNull: true,
		}
	}

	return NullableInteger{
		IsNull: false,
		Value:  *p,
	}
}

func PtrToNullString(p *string) NullableString {
	if p == nil {
		return NullableString{
			IsNull: true,
		}
	}

	return NullableString{
		IsNull: false,
		Value:  *p,
	}
}

type Task struct {
	Id          string
	Value       string
	Topics      []string
	Priority    NullableInteger
	DoneDate    NullableInteger
	DeadLine    int
	CreatedDate int
}

File: frontend/src/core/Task.ts
export class Task {
  public id: string;
  public value: string;
  public topics: string[];
  public deadLine?: Date;
  public priority: number;
  public doneDate?: Date;
  public created: Date;
  public constructor(args: {
    id: string;
    value: string;
    topics: string[];
    deadLine?: Date;
    priority?: number;
    doneDate?: Date;
    created: Date;
  }) {
    this.id = args.id;
    this.value = args.value;
    this.topics = args.topics;
    this.deadLine = args.deadLine;
    this.priority = args.priority || 0;
    this.doneDate = args.doneDate;
    this.created = args.created;
  }

  static createEmptyTask() {
    return new Task({
      id       : `${Date.now()}_${Math.random().toString(32).substring(2)}`,
      value    : '',
      deadLine : new Date(),
      created  : new Date(),
      topics   : [],
    });
  }

}